crates/blockchain/mempool.rs

```diff
@@ -18,16 +18,42 @@ use ethrex_common::{
 use ethrex_storage::error::StoreError;
 use std::collections::HashSet;
 
+// Max number of transactions in the mempool
+const MEMPOOL_MAX_SIZE: usize = 10000; //TODO: Define
+
 #[derive(Debug, Default)]
 pub struct Mempool {
     broadcast_pool: RwLock<HashSet<H256>>,
     transaction_pool: RwLock<HashMap<H256, MempoolTransaction>>,
     blobs_bundle_pool: Mutex<HashMap<H256, BlobsBundle>>,
     txs_by_sender_nonce: RwLock<BTreeMap<(H160, u64), H256>>,
+    txs_order: RwLock<Vec<H256>>,
 }
 impl Mempool {
     pub fn new() -> Self {
-        Self::default()
+        Mempool {
+            txs_order: RwLock::new(Vec::with_capacity(MEMPOOL_MAX_SIZE)),
+            ..Default::default()
+        }
+    }
+
+    /// Remove the oldest transaction in the pool
+    pub fn remove_oldest_transaction(
+        &self,
+        tx_pool: &mut std::sync::RwLockWriteGuard<'_, HashMap<H256, MempoolTransaction>>,
+    ) -> Result<(), StoreError> {
+        let mut txs_order = self
+            .txs_order
+            .write()
+            .map_err(|error| StoreError::MempoolWriteLock(error.to_string()))?;
+        let oldest_entry = txs_order.first().cloned();
+
+        if let Some(oldest_hash) = oldest_entry {
+            self.remove_transaction_with_lock(&oldest_hash, tx_pool)?;
+            txs_order.remove(0);
```

Comment:
On a `Vec`, this still requires copying all the transactions. Let's open an issue to handle this in a better way.